### PR TITLE
Introduce new preview auto-scaling mode for ARC.

### PR DESCRIPTION
**Pull request by TingluoHuang**
As mentioned in https://github.com/actions/actions-runner-controller/discussions/2072, we are trying to introduce a new auto-scaling mode to ARC and hope to improve the following areas:

- No longer require `cert-manager` as a pre-requirement for using ARC.
- Reliable scale down to zero runner pod and scale up based on job demands.
- Reduce API requests to `api.github.com` in order to avoid rate-limiting.
- No longer pass your GitHub admin privilege token to the runner pod for runner registration.
- Max flexibility on customizing your runner pod template.
- Unify ARC installation process to helm charts.

The preview auto-scaling mode introduced in this PR requires certain new APIs on GitHub.com. 
Those new APIs are behind a feature flag which is `OFF` right now, we will run through a set of testing before turning on the feature flag for everyone.

**TLDR: You can't use this new auto-scaling mode now with your GitHub enterprise/organization/repository**

### CODE REVIEW

File: charts_preview/actions-runner-controller-2/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+apiVersion: v2
+name: actions-runner-controller-2
+description: A Helm chart for install actions-runner-controller CRD
```
Comment (Link-) [inline charts_preview/actions-runner-controller-2/Chart.yaml]: We will be introducing the new OCI enabled chart publishing workflows soon, maybe it's best to avoid this folder structure change just for the sake of not triggering the existing workflows.

We can update existing release workflow easily to ignore `actions-runner-controller-2`: https://github.com/actions/actions-runner-controller/blob/3e88ae2d38b16c972c5af4fb448618767fa8d610/.github/workflows/publish-chart.yaml#L10

**Comment by Link-** [comment]
### TODO before merge

- [x] Co-author this commit with: @cory-miller @nikola-jokic @AvaStancu @fhammerl @rentziass

**Review by Link- — DISMISSED** [review]
🚀

**Review by Link- — APPROVED** [review]

**Comment by ajschmidt8** [comment]
:rocket::rocket::rocket:

looking forward to this being publicly available!!

**Comment by AlonAvrahami** [comment]
You guys rock! 🚀  can't wait! 🥇
